Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -79,18 +77,17 @@ public record DafnyDocument(
 
     public int LinesCount => VerificationTree.Range.End.Line;
     public IVerificationProgressReporter? GutterProgressReporter { get; set; }
-    public ConcurrentStack<Counterexample> Counterexamples { get; set; } = new();
-    public ConcurrentDictionary<ImplementationId, ImplementationView> ImplementationIdToView { get; set; } = new();
+    public List<Counterexample> Counterexamples { get; set; } = new();
+    public Dictionary<ImplementationId, ImplementationView> ImplementationIdToView { get; set; } = new();
```

Comment:
Why is it safe to use a regular collection here?

Reply:
It's only updated by the thread managed by `DafnyDocument.UpdateScheduler`